File: internal/auth/roles.go
package auth

// Role constants for the application
const (
	RoleMember    = "member"
	RoleAdmin     = "admin"
	RoleOrganizer = "organizer"
	RoleOwner     = "owner" // Also used in some places
)

// IsAdminRole checks if the provided role has admin privileges
func IsAdminRole(role string) bool {
	return role == RoleAdmin || role == RoleOrganizer || role == RoleOwner
}

// IsOrganizerRole checks if the provided role has organizer privileges
func IsOrganizerRole(role string) bool {
	return role == RoleOrganizer || role == RoleAdmin || role == RoleOwner
}

File: frontend/src/constants/roles.ts
// User role constants
export const ROLE_MEMBER = 'member'
export const ROLE_ADMIN = 'admin'
export const ROLE_ORGANIZER = 'organizer'
export const ROLE_OWNER = 'owner'

// Role types for TypeScript
export type Role = typeof ROLE_MEMBER | typeof ROLE_ADMIN | typeof ROLE_ORGANIZER | typeof ROLE_OWNER

// Helper functions for role checks
export const isAdminRole = (role: string): boolean => {
  return role === ROLE_ADMIN || role === ROLE_ORGANIZER || role === ROLE_OWNER
}

export const isOrganizerRole = (role: string): boolean => {
  return role === ROLE_ORGANIZER || role === ROLE_ADMIN || role === ROLE_OWNER
}
